application/positionmanager/src/positionmanager/main.py

```diff
@@ -0,0 +1,285 @@
+
+
+def get_alpaca_client():
+    return AlpacaClient(
+        api_key=os.getenv("ALPACA_API_KEY"),
+        api_secret=os.getenv("ALPACA_API_SECRET"),
+        paper=os.getenv("ALPACA_PAPER", "true").lower() == "true",
+    )
+
+
+def get_data_client():
+    return DataClient(datamanager_base_url=os.getenv("DATAMANAGER_BASE_URL"))
+
+
+def get_portfolio_optimizer():
+    return PortfolioOptimizer(
+        minimum_portfolio_tickers=int(os.getenv("MINIMUM_PORTFOLIO_TICKERS", "5")),
+        maximum_portfolio_tickers=int(os.getenv("MAXIMUM_PORTFOLIO_TICKERS", "20")),
+    )
+
+
+@application.get("/health")
+def get_health():
+    return {"status": "healthy"}
+
+
+@application.post("/positions")
+def create_position(payload: PredictionPayload) -> Dict[str, Any]:
+    alpaca_client = get_alpaca_client()
+    data_client = get_data_client()
+    portfolio_optimizer = get_portfolio_optimizer()
+
+    try:
+        cash_balance = alpaca_client.get_cash_balance()
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Error getting cash balance: {str(e)}",
+        )
+
+    end_date = datetime.now()
+    start_date = end_date - timedelta(days=trading_days_per_year)
+
+    try:
+        historical_data = data_client.get_data(start_date, end_date)
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Error getting historical data: {str(e)}",
+        )
+
+    try:
+        optimized_portfolio = portfolio_optimizer.get_optimized_portfolio(
+            historical_data, int(cash_balance)
+        )
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=500,
+            detail=f"Error optimizing portfolio: {str(e)}",
+        )
+
+    executed_trades = []
+    for ticker, weight in optimized_portfolio.items():
+        if weight <= 0:
+            continue
+
+        notional_amount = cash_balance * weight * 0.95  # 5% cash buffer
+
+        try:
+            alpaca_client.place_notional_order(ticker, notional_amount)
+
+            executed_trades.append(
+                {
+                    "ticker": ticker,
+                    "notional_amount": notional_amount,
+                    "status": "success",
+                }
+            )
+
+        except Exception as e:
+            executed_trades.append(
+                {
+                    "ticker": ticker,
+                    "notional_amount": notional_amount,
+                    "status": "error",
+                    "error": str(e),
+                }
+            )
+
+    final_cash_balance = alpaca_client.get_cash_balance()
+
+    return {
+        "status": "success",
+        "initial_cash_balance": cash_balance,
+        "final_cash_balance": final_cash_balance,
+        "optimized_portfolio": optimized_portfolio,
+        "executed_trades": executed_trades,
+        "time_period": {"start_date": start_date, "end_date": end_date},
+    }
```

Comment:
same as before on returning pydantic models

Reply:
Gonna just leave it as a dict for now but I definitely get the point.